File: 2021/01/0119js/로프.js
// const input = require("fs")
//   .readFileSync("/dev/stdin")
//   .toString()
//   .trim()
//   .split("\n");
const input = require("fs")
  .readFileSync("./2217.txt")
  .toString()
  .trim()
  .split("\n");

let N = parseInt(input[0]);
let nums = [];
for (let i = 1; i < N + 1; i++) {
  nums.push(parseInt(input[i]));
}
nums.sort((a, b) => a - b);
let Max = nums[0] * N;
for (let num of nums) {
  Max = Math.max(Max, num * N);
  N--;
}
console.log(Max);

File: 2022/09/0911/로프.py
import sys
sys.stdin = open('./text/2217.txt', 'r')
input = sys.stdin.readline
def Split():return map(int, input().strip().split())
def List():return list(map(int, input().strip().split()))
def Int():return int(input().strip())
def Str():return input().strip()

nums = [Int() for _ in range(Int())]
nums.sort(reverse=True)
Max = 0
for i in range(len(nums)):
    Max = max(Max, nums[i] * (i + 1))
print(Max)
